### PR TITLE
Support multi part namespaces

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
- Support multi-part namespaces in parsing.

### CODE REVIEW

File: ogr/parsing.py [hunk 1]
```diff
@@ -136,16 +136,21 @@ def parse_git_repo(potential_url: str) -> Optional[RepoUrl]:
         )
     if not username and len(split) >= 2:
         # path contains username/reponame
+        # or some/namespace/reponame
+        # or fork/username/some/namespace/reponame
 
-        is_fork = "fork" in split
-        if is_fork and len(split) == 4:
+        is_pagure_fork = "fork" in split
+        if is_pagure_fork and len(split) == 4:
```
Comment (mfocko) [inline ogr/parsing.py:143]: I wonder about https://gitlab.com/fork, since you can nest groups on gitlab...

File: ogr/parsing.py [hunk 2]
```diff
@@ -136,16 +136,21 @@ def parse_git_repo(potential_url: str) -> Optional[RepoUrl]:
         )
     if not username and len(split) >= 2:
         # path contains username/reponame
+        # or some/namespace/reponame
+        # or fork/username/some/namespace/reponame
 
-        is_fork = "fork" in split
-        if is_fork and len(split) == 4:
+        is_pagure_fork = "fork" in split
```
Comment (csomh) [inline ogr/parsing.py:142]: Would checking for `split[0] == fork` be better?

File: ogr/parsing.py (hunk 1 above)
Comment (csomh) [inline ogr/parsing.py:143]: Why '4'? What happens with https://pagure.io/fork/csomh/waiverdb ?

**Review by mfocko — APPROVED** [review]
I think I'll try to refactor it a bit on Friday.

File: ogr/parsing.py (hunk 2 above)
Comment (lachmanfrantisek) [reply ogr/parsing.py:142]: Good point.

Comment (lachmanfrantisek) [reply ogr/parsing.py]: Fixed.

File: ogr/parsing.py (hunk 1 above)
Comment (lachmanfrantisek) [reply ogr/parsing.py]: Yes, good point. Some historical relict from the Betka days... Fixed.

Comment (lachmanfrantisek) [reply ogr/parsing.py]: The current version checks only the first part to `fork` and https://gitlab.com/fork has not subgroups yet so it's probably safe.

File: ogr/parsing.py (hunk 2 above)
Comment (csomh) [reply ogr/parsing.py]: How about adding a test for "https://host.name/namespace/fork"? A comment saying `a repo called "fork"` would be probably useful.

File: ogr/parsing.py (hunk 1 above)
Comment (csomh) [reply ogr/parsing.py]: How about adding atest for `https://host.name/fork/username/repo`?

**Pull request merged**